data/code/lib.c

```diff
@@ -589,3 +593,732 @@ void check_lap_ex()
 #endif
 }
 #endif
+
+#define EXTENDED_TERRAIN_BOUNCE_FLAG 0x1
+#define EXTENDED_TERRAIN_BOUNCE_LIFTOFF_FLAG 0x2
+
+#define EXTENDED_TERRAIN_BOUNCY 0xB0
+
+// The importance of this remaining in its own section is that its functionality will be needed for
+// any future custom terrain types added, within the Extended Terrain Types patch or otherwise.
+#if EXTENDED_TERRAIN_TYPES
+
+typedef struct
+{
+    int point_indexes[3];
+    float unknown;
+    short normal[3];
+    char terrain_type;
+    char terrain_type_part_two;
+    char min_max_table;
+    char camera_code;
+    short neighbor_triangles[3];
+    int splash_hash;
+    // splash_hash is actually a 4-byte structure.
+
+} CollisionTriangle;
+
+struct CrsGround
+{
+    char unknown[0x20];
+    struct CollisionTriangle* col_triangle;
+};
+
+// Check against all of the custom material flags enabled by the patch.
+int is_extended_terrain_type(const char terrain_type)
+{
+    return terrain_type == EXTENDED_TERRAIN_BOUNCY;
+}
+
+int should_return_fake_code(const CollisionTriangle* triangle)
+{
+    if (triangle)
+    {
+        return is_extended_terrain_type(triangle->terrain_type);
+    }
+
+    return 0;
+}
+
+// Game will search for a Splash object due to the material hash being used.
+// This nullifies that behaviour.
+struct CollisionTriangle* get_splash_code_inline()
+{
+    register const struct CrsGround* const ground asm("r3");
+    if (should_return_fake_code(ground->col_triangle))
+    {
+        return 0;
```

Comment:
Nit: `return NULL;` would acknowledge that a pointer is being returned.

Reply:
Question: To my understanding, `NULL` is not a built-in constant within C, so what would be the expected way to satisfy this?

I could do this somewhere near the top of `lib.c`:
```c
#define NULL ((void*)0)
```

Reply:
That sounds sensible to me. There is already `true` and `false`; it can be placed next to them.